wall_e/extensions/ban.py

```diff
@@ -145,14 +145,17 @@ def pred(ban: discord.AuditLogEntry):
                          ban_date=audit_ban.created_at.timestamp(),
                          reason=audit_ban.reason if audit_ban.reason else 'No Reason Given!'
                          )
+        # unban
+        await guild.unban(member)
+        self.logger.info(f"[Ban intercept()] Guild ban for user: {ban.username} removed. Moving ban into db.")
 
         # update ban_list and db
+        success = await BanRecord.insert_record(ban)
+        if not success:
+            self.logger.info(f"[Ban intercept()] User: {member} is already banned in the system.")
```

Comment:
so the code would come here if a a mod bans a user that is already banned in the db...which I think can only happen if person A who was banned in the db rejoins the guild and a trigger-happy moderator utilizes the discord ban before the wall_e ban can fully take effect?

Reply:
no, its still possible to do a guild ban on someone thats not in the guild anymore. so this would trigger an intercept, thats why on ban listener takes a Union( member, user ) instead of just member.

Reply:
you can ban a discord user from a guild even if they are not in that guild? 👀

Reply:
> you can ban a discord user from a guild even if they are not in that guild? 👀

yes, as long as theres a reference to the user in the server. ie an clickable @ or there name in audit logs etc

Reply:
welp ok then